common/random/lcg.ksy

```diff
@@ -0,0 +1,42 @@
+meta:
+  id: lcg
+  title: Linear Congruential Generator
+  xref:
+    iso: "28640:2010"
+    wikidata: Q1190228
+  license: Unlicense
+  endian: le
+doc: |
+  A simple non-secure pseudorandom generator.
+params:
+  - id: a
+    type: u4
+  - id: b
+    type: u4
+  - id: value_bits
+    type: u1
+  - id: initial_seed
+    type: u4
```

Comment:
Technically, this implementation in 32 bits + 8-bit value is specific to Spider Solitaire. I'm not 100% sure, but may be it warrants rebadging it as `random_lcg_32_8` or something like that?